source/common/http/codec_client.cc

```diff
@@ -184,11 +184,33 @@ void CodecClient::onData(Buffer::Instance& data) {
          absl::StrCat("extraneous bytes after response complete: ", data.length()));
 }
 
+Status CodecClient::ActiveRequest::encodeHeaders(const RequestHeaderMap& headers, bool end_stream) {
+  if (header_validator_) {
+    auto result = header_validator_->transformRequestHeaders(headers);
+    if (!result.status.ok()) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("header validation failed: ", result.status.details()));
+    }
+    if (result.new_headers) {
+      std::cout << "----------------- New headers: " << std::endl
```

Comment:
Is this ready for review or do you want to do a clean up pass.
Also I tend to not do major refactors along with functional change so I'd mildly lean towards another refactor PR followed by the change, but as this code isn't in prod I'll leave it to your call
/wait on clean-up

Reply:
Should be good fore review now.